Error: The 'cache' field on 'RequestInitializerDict' is not implemented.

Started seeing this error when using AWS S3 SDK to GetObject in a Cloudflare Pages App, which runs on workerd:

```
✘ [ERROR] Error: The 'cache' field on 'RequestInitializerDict' is not implemented.

 at FetchHttpHandler.handle
  (file:///.../node_modules/@smithy/fetch-http-handler/dist-es/fetch-http-handler.js:77:30)
      at null.<anonymous> (async
```

I will attempt a downgrade. I may try a standalone app to share.

Update pinning to 3.2.5 resolves the error.

--- a/.changeset/good-cheetahs-wait.md
+++ b/.changeset/good-cheetahs-wait.md
@@ -1,0 +1,5 @@
+---
+"@smithy/fetch-http-handler": patch
+---
+
+Omit setting cache setting on request init when using default value

--- a/packages/fetch-http-handler/src/fetch-http-handler.ts
+++ b/packages/fetch-http-handler/src/fetch-http-handler.ts
@@ -112,8 +112,12 @@ export class FetchHttpHandler implements HttpHandler<FetchHttpHandlerConfig> {
       headers: new Headers(request.headers),
       method: method,
       credentials,
-      cache: this.config!.cache ?? "default",
     };
+    // cache property is not supported in workerd runtime
+    // TODO: can we feature detect support for cache and not set this property when not supported?
+    if (this.config?.cache) {
+      requestOptions.cache = this.config.cache;
+    }
 
     if (body) {
       requestOptions.duplex = "half";

